File: sys/js/array.js
export function array(n, value) {
    let a = new Array(n)
    a.fill(value)
    return a
}

export function join(list, spliter) {
    return list.join(spliter)
}

export function map(a, f) {
    return a.map(f)
}

export function filter(a, f) {
    return a.filter(f)
}

export function reduce(a, f, init) {
    return a.reduce(f, init)
}

global.array = array
global.join = join
global.reduce = reduce
global.filter = filter
global.map = map

File: sys/script1/lib/src/array.dart
List array(n, [value=null]) {
    // var a = new List(n)
    return List.filled(n, value);
}

String join(List list, String spliter) {
    return list.join(spliter);
}

List map(List a, dynamic Function(dynamic) f) {
    return a.map(f).toList();
}

List where(List a, bool Function(dynamic) f) {
    return a.where(f).toList();
}

List reduce(List a, dynamic Function(dynamic, dynamic) f, dynamic init) {
    return a.fold(init, f).toList();
}
